sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_twin.c

```diff
@@ -0,0 +1,437 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "test_az_iot_hub_client.h"
+#include <az_iot_hub_client.h>
+#include <az_span.h>
+
+#include <az_precondition.h>
+#include <az_precondition_internal.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <az_test_precondition.h>
+#include <cmocka.h>
+
+static const az_span test_device_id = AZ_SPAN_LITERAL_FROM_STR("my_device");
+static const az_span test_device_hostname = AZ_SPAN_LITERAL_FROM_STR("myiothub.azure-devices.net");
+static const az_span test_device_request_id = AZ_SPAN_LITERAL_FROM_STR("id_one");
+
+static const char test_correct_twin_response_topic_filter[] = "$iothub/twin/res/#";
+static const char test_correct_twin_get_request_topic[] = "$iothub/twin/GET/?$rid=id_one";
+static const char test_correct_twin_path_subscribe_topic[]
+    = "$iothub/twin/PATCH/properties/desired/";
+static const char test_correct_twin_patch_pub_topic[]
+    = "$iothub/twin/PATCH/properties/reported/?$rid=id_one";
+
+#ifndef NO_PRECONDITION_CHECKING
+enable_precondition_check_tests()
+
+static void test_az_iot_hub_client_twin_response_subscribe_topic_filter_get_null_client_fails(
+    void** state)
+{
+  (void)state;
+
+  uint8_t test_span_buffer[_az_COUNTOF(test_correct_twin_response_topic_filter) - 1];
+  az_span test_span = az_span_init(test_span_buffer, 0, _az_COUNTOF(test_span_buffer));
+
+  assert_precondition_checked(
```

Comment:
This name is confusing - maybe assert_precondition_failed? (Not part of this PR though it's the first time I see it.)

Reply:
Yea we could have a separate PR for changing if we want for sure.